Editor crashes when opening a Godot file in `VS Code` with the `godot-tools v2.5.1` extension .

### Tested versions

4.6 beta 2

### System information

Godot v4.6.beta2 - Windows 11 (build 26200) - Multi-window, 1 monitor - Vulkan (Mobile) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 32.0.15.9144) - AMD Ryzen 5 5600GT with Radeon Graphics (12 threads) - 15.87 GiB memory

### Issue description

When I open an godot's file (`.gd, .tscn, etc`) in VS Code with `godot-tools v2.5.1` extension, the Editor simply crashes.

After the _first crash_, the extension tries to connect to Godot again, and until it connects, I can write in the script normally until it connects and crashes again.

### Steps to reproduce

- Open a project in `Godot 4.6 beta2`
- Open a Godot's file in VS Code with `godot-tools v2.5.1` extension (The project needs to be already open)
> If the extension detects that Godot is open, it will attempt to connect, and upon succeeding, it will simply crash.

### Minimal reproduction project (MRP)

https://github.com/user-attachments/assets/06d19168-f7c6-4a81-a805-c42506d4acf6

I tried it in the following:

Rider,
Windsurf,
Codium

All would show a massive amount of errors in the code, despite showing that language server was connected successfully, then Godot would crash. 

It happened only on beta2. beta1 and 4.5 current worked fine.
I am viewing the project root, not opening a single file as a project.

System:
Godot v4.6.beta2 - KDE neon User Edition 24.04 on X11 - X11 display driver, Multi-window, 2 monitors - Vulkan (Forward+) - dedicated AMD Radeon RX 7800 XT (RADV NAVI32) - Intel(R) Core(TM) i9-14900K (32 threads) - 125.6 GiB memory

A lot of this logging looks like noise from my project I'm sorry.

**Rider verbose logging:**
```
mark@mark-gamedev:~$ /home/mark/Documents/Installs/JetBrains\ Rider-2025.2.3/bin/rider.sh
[0.003s][warning][cds] Archived non-system classes are disabled because the java.system.class.loader property is specified (value = "com.intellij.util.lang.PathClassLoader"). To use archived non-system classes, this property must not be set
Debug | ByteBufferAsyncProcessor  | 25:DefaultDispatcher-worker-8 | PAUSE ('Socket not connected') :: {id = EarlyStartServerWire/Sender, state = 'Initialized'} 
Debug | Server                    | 44:EarlyStartServerWire | EarlyStartServerWire: listening /127.0.0.1:33141 
2025-12-25 13:10:46,902 [    278]   WARN - #c.i.m.l.t.AIAssistantTelemetryExporterKt - No Langfuse auth string found.
Installation home directory: file:////home/mark/Documents/Installs/JetBrains Rider-2025.2.3
System directory: file:////home/mark/.cache/JetBrains/Rider2025.2
Config directory: file:////home/mark/.config/JetBrains/Rider2025.2
Log directory: file:////home/mark/.cache/JetBrains/Rider2025.2/log
2025-12-25 13:10:47,253 [    629]   WARN - #c.i.s.c.i.StationSocketConnectionLoop - Discovery connection failed with NoSuchFileException (/run/user/1000/jb.station.sock)
2025-12-25 13:10:47,422 [    798]   WARN - #c.i.s.c.i.StationSocketConnectionLoop - RecentProjects connection failed with NoSuchFileException (/run/user/1000/jb.station.sock)
2025-12-25 13:10:57,045 [  10421]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/icons/gdshader.svg, icon.url.path=file:/home/mark/.local/share/JetBrains/Rider2025.2/gdshader/lib/gdshader-0.3.1-beta.jar!/icons/gdshader.svg
2025-12-25 13:10:57,046 [  10422]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/icons/gdshader.svg, icon.url.path=file:/home/mark/.local/share/JetBrains/Rider2025.2/gdshader/lib/gdshader-0.3.1-beta.jar!/icons/gdshader.svg
2025-12-25 13:10:57,153 [  10529]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/unityIcons/debugger/db_no_suspend_pausepoint.svg, icon.url.path=file:/home/mark/Documents/Installs/JetBrains%20Rider-2025.2.3/plugins/rider-unity/lib/rider-plugins-unity.jar!/unityIcons/debugger/db_no_suspend_pausepoint.svg
2025-12-25 13:10:57,154 [  10530]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/unityIcons/debugger/db_disabled_pausepoint.svg, icon.url.path=file:/home/mark/Documents/Installs/JetBrains%20Rider-2025.2.3/plugins/rider-unity/lib/rider-plugins-unity.jar!/unityIcons/debugger/db_disabled_pausepoint.svg
2025-12-25 13:10:57,154 [  10530]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/unityIcons/debugger/db_muted_pausepoint.svg, icon.url.path=file:/home/mark/Documents/Installs/JetBrains%20Rider-2025.2.3/plugins/rider-unity/lib/rider-plugins-unity.jar!/unityIcons/debugger/db_muted_pausepoint.svg
2025-12-25 13:10:57,155 [  10531]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/unityIcons/debugger/db_muted_disabled_pausepoint.svg, icon.url.path=file:/home/mark/Documents/Installs/JetBrains%20Rider-2025.2.3/plugins/rider-unity/lib/rider-plugins-unity.jar!/unityIcons/debugger/db_muted_disabled_pausepoint.svg
2025-12-25 13:10:57,155 [  10531]   WARN - #c.j.r.u.i.c.ImageIconConverter - creating model without coords: originalPath=/unityIcons/debugger/db_dep_line_pausepoint.svg, icon.url.path=file:/home/mark/Documents/Installs/JetBrains%20Rider-2025.2.3/plugins/rider-unity/lib/rider-plugins-unity.jar!/unityIcons/debugger/db_dep_line_pausepoint.svg
2025-12-25 13:10:57,163 [  10539]   WARN - #c.i.m.l.c.q.QuotaManager2Impl - Quota refill state was updated by another thread, wanted to update to Error(exception=com.intellij.ml.llm.core.quota.ResultDoesNotMatchConditionException: Returned result does not match the condition, previous=Known(next=2026-01-06T10:00:08.844Z, tariff=QuotaRefillInfoTariff(amount=3500000, duration=30d)))
2025-12-25 13:10:57,175 [  10551]   WARN - #c.i.m.l.c.m.LlmProfileService - third party provider Ollama: failed to get available llm profiles
java.net.ConnectException
        at java.net.http/jdk.internal.net.http.common.Utils.toConnectException(Utils.java:1066)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:227)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.checkRetryConnect(PlainHttpConnection.java:280)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$2(PlainHttpConnection.java:238)
        at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
        at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
        at kotlinx.coroutines.internal.SoftLimitedDispatcher$Worker.run(SoftLimitedDispatcher.kt:130)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:613)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:1183)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:778)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:765)
Caused by: java.nio.channels.ClosedChannelException
        at java.base/sun.nio.ch.SocketChannelImpl.ensureOpen(SocketChannelImpl.java:202)
        at java.base/sun.nio.ch.SocketChannelImpl.beginConnect(SocketChannelImpl.java:786)
        at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:874)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$1(PlainHttpConnection.java:210)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:212)
        ... 12 more
2025-12-25 13:10:59,026 [  12402] SEVERE - #c.i.p.i.s.PsiSearchHelperImpl - Error during processing of: mesh_baker.gd
com.intellij.psi.stubs.UpToDateStubIndexMismatch: PSI and index do not match.
Please report the problem to JetBrains with the files attached
INDEXED VERSION IS THE CURRENT ONE file=GdScript File, file.class=class gdscript.psi.GdFile, file.lang=Language: GdScript, modStamp=1, psi.length=12164
 tree consistent
 stub debugInfo=created in getStubTree(), with AST = false; with backReference
 viewProvider=com.intellij.psi.SingleRootFileViewProvider{vFile=file:///media/mark/GameDevelopment/project/addons/zylann.hterrain/tools/mesh_baker.gd, vFileId=51197, content=VirtualFileContent{size=12165}, eventSystemEnabled=true}
 viewProvider stamp: 0; file stamp: 0; file modCount: 1762664634852; file length: 12165
 doc saved: true; doc stamp: 0; doc size: 12164; committed: true
indexing info: indexing timestamp = 1762664634852, binary = false, byte size = 12165, char size = 12164
ref: 20250127
latestIndexedStub=StubTree{myDebugInfo='created from index; with backReference', myRoot=PsiFileStubImpl}852283482
   same size=true
   debugInfo=created from index; with backReference
        at com.intellij.psi.stubs.StubTreeLoader.handleUpToDateMismatch(StubTreeLoader.java:224)
        at com.intellij.psi.stubs.StubTreeLoader.access$100(StubTreeLoader.java:25)
        at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.doCreateCompleteException(StubTreeLoader.java:216)
        at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.lambda$createCompleteException$0(StubTreeLoader.java:192)
        at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:756)
        at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:712)
        at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:298)
        at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:139)
        at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:298)
        at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.createCompleteException(StubTreeLoader.java:191)
        at com.intellij.psi.impl.source.PsiFileImpl.loadTreeElement(PsiFileImpl.java:243)
        at com.intellij.psi.impl.source.PsiFileImpl.calcTreeElement(PsiFileImpl.java:844)
        at com.intellij.psi.impl.source.PsiFileImpl.getNode(PsiFileImpl.java:1043)
        at com.intellij.psi.impl.source.PsiFileImpl.getNode(PsiFileImpl.java:55)
        at com.intellij.psi.impl.search.LowLevelSearchUtil.processElementsAtOffsets(LowLevelSearchUtil.java:173)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl$6.execute(PsiSearchHelperImpl.java:1107)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.processInReadAction(PsiSearchHelperImpl.java:886)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.processInReadAction(PsiSearchHelperImpl.java:876)
        at com.intellij.openapi.application.ReadActionProcessor.lambda$process$0(ReadActionProcessor.java:11)
        at com.intellij.openapi.application.impl.AppImplKt$rethrowCheckedExceptions$2.invoke(appImpl.kt:106)
        at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runReadAction(NestedLocksThreadingSupport.kt:784)
        at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:1043)
        at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
        at com.intellij.openapi.application.ReadActionProcessor.process(ReadActionProcessor.java:11)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processVirtualFile$13(PsiSearchHelperImpl.java:657)
        at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:826)
        at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1221)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl.processVirtualFile(PsiSearchHelperImpl.java:611)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processPsiFileRoots$8(PsiSearchHelperImpl.java:472)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$10(PsiSearchHelperImpl.java:553)
        at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:258)
        at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$11(PsiSearchHelperImpl.java:552)
        at com.intellij.concurrency.ApplierCompleter.processArrayItem(ApplierCompleter.java:121)
        at com.intellij.concurrency.ApplierCompleter.processArray(ApplierCompleter.java:219)
        at com.intellij.concurrency.ApplierCompleter.helpOthers(ApplierCompleter.java:239)
        at com.intellij.concurrency.ApplierCompleter.helpAll(ApplierCompleter.java:181)
        at com.intellij.concurrency.ApplierCompleter.lambda$exec$0(ApplierCompleter.java:111)
        at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:826)
        at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1221)
        at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$2(ApplierCompleter.java:153)
        at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:681)
        at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:756)
        at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:712)
        at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:680)
        at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
        at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:164)
        at com.intellij.concurrency.ApplierCompleter.wrapAndRun(ApplierCompleter.java:148)
        at com.intellij.concurrency.ApplierCompleter.exec(ApplierCompleter.java:113)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.AssertionError: Stub count (16) doesn't match stubbed node length (86)
        at com.intellij.psi.impl.source.FileTrees.lambda$reconcilePsi$4(FileTrees.java:188)
        at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:561)
        at com.intellij.psi.impl.source.FileTrees.reconcilePsi(FileTrees.java:176)
        at com.intellij.psi.impl.source.FileTrees.withAst(FileTrees.java:142)
        at com.intellij.psi.impl.source.PsiFileImpl.loadTreeElement(PsiFileImpl.java:229)
        ... 42 more
2025-12-25 13:10:59,027 [  12403] SEVERE - #c.i.p.i.s.PsiSearchHelperImpl - JetBrains Rider 2025.2.5  Build #RD-252.28238.13
2025-12-25 13:10:59,027 [  12403] SEVERE - #c.i.p.i.s.PsiSearchHelperImpl - JDK: 21.0.9; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.
2025-12-25 13:10:59,027 [  12403] SEVERE - #c.i.p.i.s.PsiSearchHelperImpl - OS: Linux
2025-12-25 13:10:59,027 [  12403] SEVERE - #c.i.p.i.s.PsiSearchHelperImpl - Last Action: 
2025-12-25 13:11:02,359 [  15735]   WARN - #c.i.m.l.c.m.LlmProfileService - Suppressed a frequent exception logged for the 2nd time: null
```

No Godot logs at all. I ran with the --verbose tag. It showed all of the start up logs etc at very fine detail, but then not a single log after I opened rider.

These are the last few lines:

```
Generated 'res://WindAndWeatherSystem/Shaders/Rain/terrain_rainy.tres' preview in 15 usec
Generated 'res://WindAndWeatherSystem/Shaders/StandardCutOut/CutOutShaderSimple.gdshader' preview in 3 usec
Generated 'res://WindAndWeatherSystem/Shaders/vegetation.gdshader' preview in 1 usec
Generated 'res://WindAndWeatherSystem/Shaders/vegetation_visual.tres' preview in 13 usec
Generated 'res://WindAndWeatherSystem/fog_volume.tscn' preview in 3 usec
Generated 'res://WindAndWeatherSystem/rain_part_custom.gdshader' preview in 2 usec
Generated 'res://WindAndWeatherSystem/rain_system.tscn' preview in 3 usec
Generated 'res://WindAndWeatherSystem/WindAndWeatherController.tscn' preview in 3 usec
Generated 'res://default_bus_layout.tres' preview in 13 usec
Generated 'res://e2e.log' preview in 18 usec
Generated 'res://export_presets.cfg' preview in 12 usec
Generated 'res://gamecontrollerdb.txt' preview in 11 usec
Generated 'res://README.md' preview in 12 usec
Generated 'ID:-9223369635598872459' preview in 53 usec
```


Obviously unrelated, but just to demonstrate what I mean.

--- a/modules/gdscript/language_server/gdscript_language_protocol.cpp
+++ b/modules/gdscript/language_server/gdscript_language_protocol.cpp
@@ -389,12 +389,15 @@ ExtendGDScriptParser *GDScriptLanguageProtocol::LSPeer::parse_script(const Strin
 	}
 
 	ExtendGDScriptParser *parser = memnew(ExtendGDScriptParser);
+	parse_results[p_path] = parser;
+
 	parser->parse(content, p_path);
 
 	if (document != nullptr) {
-		parse_results[p_path] = parser;
 		GDScriptLanguageProtocol::get_singleton()->get_workspace()->publish_diagnostics(p_path);
 	} else {
+		// Don't keep cached for further requests since we can't invalidate the cache properly.
+		parse_results.erase(p_path);
 		stale_parsers[p_path] = parser;
 	}
 

